lib/parser/attribute-parser.ts

```diff
@@ -1,19 +1,72 @@
-import { OAttribute, ObjectBase } from "./objects";
+import { OLexerToken } from "../lexer";
+import { ExpressionParser } from "./expression-parser";
+import { OAttributeDeclaration, OAttributeSpecification, ObjectBase, OReference, ParserError } from "./objects";
 import { ParserBase, ParserState } from "./parser-base";
 
 export class AttributeParser extends ParserBase {
-  constructor(state: ParserState, private parent: ObjectBase ) {
+  constructor(state: ParserState, private parent: ObjectBase) {
     super(state);
     this.debug(`start`);
 
   }
-  parse(): OAttribute {
-    this.expect('attribute');
-    const attribute = new OAttribute(this.parent, this.state.pos.getRangeToEndLine());
-    attribute.lexerToken = this.consumeToken();
-    this.advanceSemicolon(true);
+  parse() {
+    const attribute = this.expect('attribute');
+    const token = this.consumeToken();
+    if (this.getToken().getLText() === 'of') {
+      return this.parseAttributeSpecification(token, attribute);
+    } else if (this.getToken().getLText() === ':') {
+      return this.parseAttributeDeclaration(token, attribute);
+    } else {
+      throw new ParserError(`Unexpected token ${this.getToken().text} in AttributeParser`, this.getToken().range);
```

Comment:
Add that we expected a `:` or `of`.

And wouldn't it be possible to recover in the most cases by throwing away stuff until the next `;`? Maybe only when it's less than 10 tokens away or something like that?
But maybe that's a bit arbitrary...